curiosrc/seal/task_treed.go

```diff
@@ -216,19 +232,31 @@ func (t *TreeDTask) Do(taskID harmonytask.TaskID, stillOwned func() bool) (done
 
 					closers = append(closers, pr)
 
-					pieceReaders[i], _ = padreader.New(pr, uint64(*p.DataRawSize))
+					reader, _ := padreader.New(pr, uint64(*p.DataRawSize))
+					pieceReaders = append(pieceReaders, reader)
 				} else {
-					pieceReaders[i], _ = padreader.New(&UrlPieceReader{
+					reader, _ := padreader.New(&UrlPieceReader{
 						Url:     dataUrl,
 						RawSize: *p.DataRawSize,
 					}, uint64(*p.DataRawSize))
+					pieceReaders = append(pieceReaders, reader)
 				}
 
 			} else { // padding piece (w/o fr32 padding, added in TreeD)
-				pieceReaders[i] = nullreader.NewNullReader(abi.PaddedPieceSize(p.PieceSize).Unpadded())
+				pieceReaders = append(pieceReaders, nullreader.NewNullReader(abi.PaddedPieceSize(p.PieceSize).Unpadded()))
 			}
 		}
 
+		if offset.Padded() < abi.PaddedPieceSize(ssize) {
+			fillerSize := abi.PaddedPieceSize(ssize) - offset.Padded()
+			filler := nullreader.NewNullReader(fillerSize.Unpadded())
+			pieceInfos = append(pieceInfos, abi.PieceInfo{
+				Size:     fillerSize,
+				PieceCID: zerocomm.ZeroPieceCommitment(fillerSize.Unpadded()),
+			})
+			pieceReaders = append(pieceReaders, filler)
```

Comment:
The suffix fillers must be power-of-two sized; See https://github.com/filecoin-project/lotus/blob/master/storage/pipeline/states_sealing.go#L91-L133

Reply:
I already fixed this locally last night after the commit. It still produces the same error.